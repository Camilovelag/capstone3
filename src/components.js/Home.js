import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { searchUpdate } from '../redux/reducer/countriesSlice';
import '../App.css';

const img = require('../assets/continent.png');

const Home = () => {
  const country = useSelector((state) => state.countries.countries);
  const search = useSelector((state) => state.countries.search);
  const dispatch = useDispatch();
  const handleChange = (e) => {
    dispatch(searchUpdate(e.target.value));
  };

  return (
    <div>
      <div className="continent">
        <div className="continent-img">
          <img src={img} alt="continent" />
        </div>
        <div className="continent-text">
          <h1 className="continent-title">South America</h1>
        </div>
      </div>
      <div className="inputContainer">
        <input value={search} type="text" onChange={handleChange} className="inputTex" />
      </div>
      <div className="container">
        {
          search === ''
            ? country.map((item) => (
              <div className="card" key={item.name}>
                <div className="card-img">
                  <img
                    src={item.flag
                      ? item.flag
                      : 'https://www.countryflags.io/xx/flat/64.png'}
                    alt="flag"
                    className="flag"
                  />
                </div>
                <div className="card-text">
                  <h2 className="card-title">{item.name}</h2>
                </div>
              </div>
            ))
            : country.filter((item) => item.name
              .toLowerCase()
              .includes(search.toLowerCase())).map((item) => (
                <div className="card" key={item.name}>
                  <div className="card-img">
                    <img
                      src={item.flag
                        ? item.flag
                        : 'https://www.countryflags.io/xx/flat/64.png'}
                      alt="flag"
                      className="flag"
                    />
                  </div>
                  <div className="card-text">
                    <h2 className="card-title">{item.name}</h2>
                  </div>
                </div>
            ))
        }
      </div>
    </div>
  );
};

export default Home;
