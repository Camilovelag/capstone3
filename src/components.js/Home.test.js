import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Home from './Home';

describe('Home', () => {
  test('renders Home component', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Router>
          <Home />
        </Router>
      </Provider>,
    );
    expect(component).toMatchSnapshot();
  });
});
