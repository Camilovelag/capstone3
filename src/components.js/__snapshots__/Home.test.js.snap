// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders Home component 1`] = `
<div>
  <div
    className="inputContainer"
  >
    <input
      className="inputTex"
      onChange={[Function]}
      placeholder="<      country      >"
      type="text"
      value=""
    />
  </div>
  <div
    className="Continent"
  >
    <div>
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Continent"
          src="continent.png"
        />
      </a>
    </div>
    <div>
      <h2>
        South America
      </h2>
    </div>
  </div>
  <div
    className="country-title-cont"
  >
    <span
      className="country-title"
    >
      COUNTRIES
    </span>
  </div>
  <hr
    className="hr"
  />
  <div
    className="container"
  />
</div>
`;
